File: Trabalhos Práticos/Palindromo.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

void isPalindromo(char palavra[], int tamanho);

int main() {
    char palavra[100];
    
    /* Usei a estrutura while para chamar o método e rodar até que a palavra digitada seja "FIM" */
    while (true) {
        fgets(palavra, 100, stdin);
        palavra[strcspn(palavra, "\n")] = '\0'; /* Uso essa função para remover o caracter de nova linha e nao dar erro*/
        
        if (strcmp(palavra, "FIM") == 0) {
           break;
        }
        
        int tamanho = strlen(palavra);
        isPalindromo(palavra, tamanho);
    }
    
    return 0;
}

// Função para verificar se a string é ou nao um palindrmo //
void isPalindromo(char palavra[], int tamanho) {
    bool resp = true;

    for (int i = 0; i < tamanho; i++) {
        if (palavra[i] == palavra[tamanho - i -1]) {

        } else resp = false;
    }

    if (resp == true) {
        printf("SIM\n");
    } else printf("NAO\n");
}

File: Trabalhos Práticos/Palindromo.java
public class Palindromo {
    public static void main (String[] args) {
        String palavra = MyIO.readLine();
        int tamanho = palavra.length();
        // Usei a estrutura while para chamar o método e rodar até que a palavra digitada seja "FIM" //
        while (!fim(palavra)) {
            isPalindromo(palavra, tamanho);
            palavra = MyIO.readLine();
            tamanho = palavra.length();
        }
    }
       
    // Método para verificar se a palavra inserida é ou não um palíndromo //
    public static void isPalindromo(String palavra, int tamanho) { 
        boolean resp = true;
        
        for (int i = 0; i < palavra.length(); i++) {
            if (palavra.charAt(i) == palavra.charAt(tamanho - i - 1)) {

            }  else resp = false;
        }
        
        if (resp == true) {
            MyIO.println("SIM");
        } else MyIO.println("NAO");
        
    }
        
    // Método para verificar se a palavra digitada foi "FIM", e caso seja, parar o código //
    public static boolean fim (String palavra) {
        String fim = "FIM";
        if (palavra.length() != fim.length()) {
            return false;
        }

        for (int i = 0; i < fim.length(); i++) {
            if (palavra.charAt(i) != fim.charAt(i)) {
                return false;
            }
        }

        return true;
    }
}
